packages/cli/src/util/link/handle-git-connection.ts

```diff
@@ -0,0 +1,127 @@
+import { Dictionary } from '@vercel/client';
+import { parseRepoUrl } from '../git/connect-git-provider';
+import { Output } from '../output';
+import Client from '../client';
+import { Org, Project, ProjectSettings } from '../../types';
+import { handleOptions } from './handle-options';
+import {
+  promptGitConnectMultipleUrls,
+  promptGitConnectSingleUrl,
+} from './git-connect-prompts';
+
+function getProjectSettings(project: Project): ProjectSettings {
+  return {
+    createdAt: project.createdAt,
+    framework: project.framework,
+    devCommand: project.devCommand,
+    installCommand: project.installCommand,
+    buildCommand: project.buildCommand,
+    outputDirectory: project.outputDirectory,
+    rootDirectory: project.rootDirectory,
+    directoryListing: project.directoryListing,
+    nodeVersion: project.nodeVersion,
+    skipGitConnectDuringLink: project.skipGitConnectDuringLink,
+  };
+}
+
+export async function handleGitConnection(
+  client: Client,
+  org: Org,
+  output: Output,
+  project: Project,
+  remoteUrls: Dictionary<string>,
+  settings?: ProjectSettings
+): Promise<number | void> {
```

Comment:
Out of curiosity, why does this return a number? I know `connectGitProvider()` is where the number originates on error, but it's not clear what `/v4/projects/{id}/link` returns or why an `Error` isn't thrown instead with a message and code.

Reply:
There is a convention of returning numbers up a call stack to use as an exit code, but I agree that it's probably not one we want to perpetuate. There have been discussions before about wanting to refactor this convention away. I think we can refine this in a different PR.

Reply:
Related/unrelated, is there a reason we haven't upgraded to v9 of the api and secondly are there api docs for v4?

Reply:
I think the only reason I used `v4` was because I was looking at code in `front` that used `v4`. I just updated it to use `v9` and tested and it appears to work!